hal/common/FileBase.cpp

```diff
@@ -18,24 +18,24 @@
 namespace mbed {
 
 FileBase *FileBase::_head = NULL;
-PlatformMutex FileBase::_mutex;
+SingletonPtr<PlatformMutex> FileBase::_mutex;
 
 FileBase::FileBase(const char *name, PathType t) : _next(NULL),
                                                    _name(name),
                                                    _path_type(t) {
-    _mutex.lock();
+    _mutex->lock();
```

Comment:
RAII would be good to have, as it would prevent the mutex from accidentally being left locked. It its not directly related to this PR though.  Feel free to add a mutex RAII utility class in another PR.